sklearn/utils/_array_api.py

```diff
@@ -604,6 +604,14 @@ def _add_to_diagonal(array, value, xp):
             array[i, i] += value
 
 
+def max_supported_float_precision(xp, device):
+    """Return the float dtype with the highest precision supported by the device."""
```

Comment:
```suggestion
    """Return the float dtype with the highest precision supported by the device."""
    # TODO: Update to use `__array_namespace__info__()` from array-api v2023.12
    # when/if that becomes more widespread.
```

Reply:
Maybe also rename with a leading `_` to avoid having this involuntarily exposed a public attribute of a public module where it is important (although that is quite unlikely, since those public modules should all define `__all__` attribute to make their public API explicit).